src/lib.rs

```diff
@@ -349,9 +355,20 @@ pub fn run() -> Result<ExitStatus> {
             }
 
             let available_targets = rustup::available_targets(&toolchain, verbose)?;
-            let uses_xargo = config
-                .xargo(&target)?
-                .unwrap_or_else(|| !target.is_builtin() || !available_targets.contains(&target));
+            let uses_xargo = config.xargo(&target)?.unwrap_or(!target.is_builtin());
+            let uses_build_std = match uses_xargo {
+                true => false,
+                false => {
+                    !available_targets.contains(&target)
+                        || needs_build_std(&toolchain, &target, verbose)?
+                }
+            };
+            if !is_nightly && uses_build_std {
+                eyre::bail!(
+                    "no rust-std component available for {}: must use nightly",
+                    target.triple()
+                );
+            }
```

Comment:
what if we're `no_std`?

Reply:
> what if we're `no_std`?

Uh true, so then just disable it? I guess we should just let cargo take care of it then. I should probably also add a changelog entry. I should probably also check if `rust-src` is available.... Hmmm.